application/controllers/UserManagementController.php

```diff
@@ -228,16 +228,85 @@ public function actionDeleteConfirm()
         }
         $userId = Yii::app()->request->getPost('userid');
         if ($userId == Yii::app()->user->id) {
-            Yii::app()->setFlashMessage(gT("you cannot delete yourself."), 'error');
-            $this->redirect(Yii::app()->createUrl("userManagement/index"));
+            return App()->getController()->renderPartial('/admin/super/_renderJson', [
+                'data' => [
+                    'success' => false,
+                    'message' => gT("you cannot delete yourself.")
+                ]
+            ]);
+        }
+
+        $message = '';
+        $transferTo = Yii::app()->request->getPost('transfer_surveys_to');
+        
+        if (is_null($transferTo)) {
+            // If $transferTo is null, check if user owns a survey. 
+            // If so, render the "transfer to" selection screen
+            $aOwnedSurveys = Survey::model()->findAllByAttributes(array('owner_id' => $userId));
+            if (count($aOwnedSurveys)) {
+                $postuser = flattenText(Yii::app()->request->getPost("user"));
+                $aUsers = User::model()->findAll();
+                return Yii::app()->getController()->renderPartial(
+                    '/admin/super/_renderJson',
+                    [
+                        "data" => [
+                            'success' => true,
+                            'html' => $this->renderPartial(
+                                'partial/transfersurveys',
+                                [
+                                    'postuserid' => $userId,
+                                    'postuser' => $postuser,
+                                    'current_user' => Yii::app()->user->id,
+                                    'users' => $aUsers,
+                                ],
+                                true
+                            ),
+                        ]
+                    ]
+                );
+            }
+        } else {
+            // If $transferTo is not null, transfer the surveys
```

Comment:
If $transferTo is not null but false or empty ? It's a dropdown to select it ?

Reply:
Fixed it. now checks for empty before.
Yes, it is a dropdown to select it. That's on the view.